src/overlays/actors/ovl_En_Ending_Hero5/z_en_ending_hero5.c

```diff
@@ -21,18 +23,91 @@ const ActorInit En_Ending_Hero5_InitVars = {
     (ActorFunc)EnEndingHero5_Update,
     (ActorFunc)EnEndingHero5_Draw
 };
-*/
 
-GLOBAL_ASM("asm/non_matchings/ovl_En_Ending_Hero5_0x80C238C0/EnEndingHero5_Init.asm")
+extern SkeletonHeader D_0600A850;
+extern AnimationHeader D_06002FA0;
+
+extern Gfx D_060070C0[];
+extern Gfx D_06006FB0[];
+extern Gfx D_06006E80[];
+extern Gfx D_06006D70[];
+extern Gfx D_0600A390[];
+
+void EnEndingHero5_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnEndingHero5* this = THIS;
+
+    this->actor.unkA0.mass = 0xFF;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.unk1F = 6;
+    this->actor.gravity = -3.0f;
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600A850, &D_06002FA0, this->limbDrawTable,
+                     this->transitionDrawTable, 17);
+    Actor_SetDrawParams(&this->actor.shape, 0.0f, func_800B3FC0, 25.0f);
+    this->unk25C = this->actor.params;
+    func_80C23980(this);
+}
+
+void EnEndingHero5_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_80C23980(EnEndingHero5* this) {
+    this->unk258 = 1;
+    this->actionFunc = func_80C2399C;
+}
+
+void func_80C2399C(EnEndingHero5* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+void EnEndingHero5_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnEndingHero5* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    func_800B78B8(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1D);
+}
+
+Gfx* D_80C23BF0[] = { D_060070C0, D_06006FB0, D_06006E80, D_06006D70, D_0600A390 };
+
+void func_80C23A30(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnEndingHero5* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if (limbIndex == 15) {
+        gSPDisplayList(oGfxCtx->polyOpa.p++, D_80C23BF0[this->unk25C]);
+    }
 
-GLOBAL_ASM("asm/non_matchings/ovl_En_Ending_Hero5_0x80C238C0/EnEndingHero5_Destroy.asm")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
 
-GLOBAL_ASM("asm/non_matchings/ovl_En_Ending_Hero5_0x80C238C0/func_80C23980.asm")
+void EnEndingHero5_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnEndingHero5* this = THIS;
 
-GLOBAL_ASM("asm/non_matchings/ovl_En_Ending_Hero5_0x80C238C0/func_80C2399C.asm")
+    func_8012C28C(globalCtx->state.gfxCtx);
+    func_8012C2DC(globalCtx->state.gfxCtx);
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-GLOBAL_ASM("asm/non_matchings/ovl_En_Ending_Hero5_0x80C238C0/EnEndingHero5_Update.asm")
+    switch (this->unk25C) {
+        case 0:
+            gDPSetEnvColor(oGfxCtx->polyOpa.p++, 0xAA, 0x0A, 0x46, 0xFF);
+            break;
```

Comment:
mm was involved with the deimal color vote a while back so i imagine that should be used here too.
you can use -dc in gfxdis to get decimal colors automatically